Lib/tarfile.py

```diff
@@ -2345,6 +2345,15 @@ def next(self):
                     raise ReadError(str(e))
             except SubsequentHeaderError as e:
                 raise ReadError(str(e))
+            except Exception as e:
+                try:
+                    import zlib
+                    if isinstance(e, zlib.error):
+                        raise ReadError(f'zlib error: {e}')
```

Comment:
Shouldn't this have `from e` at the end?

Reply:
Please read the description of the backport.